torch/nn/modules/module.py

```diff
@@ -571,7 +571,8 @@ def compute_should_use_set_data(tensor, tensor_applied):
                         param.grad.data = grad_applied
                     else:
                         assert param.grad.is_leaf
-                        self._parameters[key].grad = grad_applied.requires_grad_(param.grad.requires_grad)
+                        self._parameters[key].grad = grad_applied.requires_grad_(  # type: ignore[union-attr]
```

Comment:
Why is this ignore? 
there is a `if param.grad is not None` just above so I guess it is not for the `.grad` field?

Reply:
python/mypy#10784

Reply:
python/mypy#10784

Reply:
Ho very good point from them.
I guess we should replace the `self._parameters[key]` with `param` here then?

Reply:
Actually as `.items()` gives reference to an object, this is a very good point, updated.